### PR TITLE
Cherry-picks: Run integration tests in parallel

### PR DESCRIPTION
**Pull request by zjs**
Cherry-pick:
* Scope the installed version of VIC in upgrade tests, so that parallel testing succeeds (f9a1600663da680b9763ce9d1db37ea24bdabb1b from #7972)
* Add basic integration tests for VCH inspect API (21f713d42dd6d8836f68e273225eac33c7e6df28 from #7290)
* Enable Group 23 - Vic Machine Service to run in parallel under CI (280704fb11f69fea5c10687a73aa470cf6b4ff62 from #7961)
* Provide per-suite vic-machine-server logs in CI (346982a76e5947d62b6629c6826223a84881640c from #7978)
* Improve docker cp test concurrency (e465c53a04a5124a1b7626d34ccde9840c8a35bb from #7987)
* Address spreading use of secret tag in our curl commands (050999cf412a7488c039e7315472222cfa4ebfce from #7862)
* Remove errant file that accidentally got merged (718f43025dbf0e20f217c26c9351c992e73d72b2 from #7873)
* Scope admin keywords so that they can be run in parallel (6e9a91ad4691d38a43050fc07701a042d4b0e458 from #8005)
* Scope the BC test so that it can run in parallel (0ce4c760dd47cf06e8acbb8d3e2b52f2538eb631 from #8004)
* Fix the for loop in the concurrent rm test which was never working (81d476243a374c6f0a6832fddec29dd3a69b1c0c from #8017)
* Parallelize integration tests (02ba3ad0b396134d0e51af2885a2414900a3f6b3 from #5601)

---

`[full ci]`

### CODE REVIEW

File: tests/test-cases/Group1-Docker-Commands/1-06-Docker-Run.robot [hunk 1]
```diff
@@ -239,5 +239,7 @@ Docker run --rm concurrent
     \   Log  ${res.stdout}
     \   Should Be Equal As Integers  ${res.rc}  0
 
+    Sleep  3 minutes
```
Comment (hickeng) [inline tests/test-cases/Group1-Docker-Commands/1-06-Docker-Run.robot]: we could instead use a docker event stream for deletions with a timeout so we don't wait longer than necessary. That said, not suggesting changing it in this PR.

File: tests/test-cases/Group12-VCH-BC/12-01-Delete.robot [hunk 2]
```diff
@@ -19,18 +19,17 @@ Suite Setup  Install VIC 1.1.1 to Test Server
...
 *** Keywords ***
-Clean up VIC Appliance And Local Binary
-    Cleanup VIC Appliance On Test Server
-    Run  rm -rf vic.tar.gz vic
-
 Install VIC 1.1.1 to Test Server
     Log To Console  \nDownloading VIC 1.1.1 from gcp...
-    ${rc}  ${output}=  Run And Return Rc And Output  wget https://storage.googleapis.com/vic-engine-releases/vic_1.1.1.tar.gz -O vic.tar.gz
-    ${rc}  ${output}=  Run And Return Rc And Output  tar zxvf vic.tar.gz
+    ${val}=  Evaluate  time.clock()  modules=time
```
Comment (hickeng) [inline tests/test-cases/Group12-VCH-BC/12-01-Delete.robot]: Seems as though this should be using `Install VIC with version to Test Server` but maybe there's a reason? Again, not something that needs addressing in this PR.

File: VCH-0-1212/VCH-0-1212.env [hunk 3]
```diff
@@ -0,0 +1 @@
+DOCKER_HOST=10.170.123.186:2376
```
[the remark this answers is not in the document]
Comment (zjs) [reply VCH-0-1212/VCH-0-1212.env]: Addressed this by cherry-picking 718f430 from #7873.

[the remark this answers is not in the document]
Comment (zjs) [reply VCH-0-1212/VCH-0-1212.env]: Created #8170 to avoid this sort of issue in the future.

**Pull request merged**